lib/wibox/widget/imagebox.lua

```diff
@@ -79,36 +81,48 @@ end
 --- Set an imagebox' image
 -- @property image
 -- @param image Either a string or a cairo image surface. A string is
---   interpreted as the path to a png image file.
+--   interpreted as the path to a image file.
 -- @return true on success, false if the image cannot be used
-
 function imagebox:set_image(image)
     if type(image) == "string" then
-        image = surface.load(image)
-        if not image then
-            print(debug.traceback())
-            return false
+        -- try to load rsvg handle from file if it's vector
+        local handle, width, height = try_to_load_svg(image)
```

Comment:
With my comment above, `try_to_load_svg` would only have to return the handle and not also get the dimensions. I feel like this then really does not need to be a separate function and can just be inlined here. Do you agree?

Reply:
>try_to_load_svg would only have to return the handle and not also get the dimensions

Theoretically rsvg can load image with zero width or height. According current surface handling it should count as 'load failed'.  So we have to mess around `handle:get_dimensions()` in the `set_image` body. Or modify helper to return validation result, something like
```lua
local function set_handle(ib, handle)
    local dim = handle:get_dimensions()
    local is_handle_valid = dim.width > 0 and dim.height > 0
    if not is_handle_valid then return end
    
    ib._private.default = { width = dim.width, height = dim.height }
    ib._private.handle = handle
    ib._private.image = nil
    return true
end

-- somewhere inside 'imagebox:set_image'
local rsvg_succeed
local handle, err = Rsvg.Handle.new_from_file(image)
if not err then
	rsvg_succeed = set_handle(self, handle)
end

if not rsvg_succeed then
	-- rsvg handle failed, try to load cairo surface with pixbuf
end
```
Is this approach ok?

Reply:
> Theoretically rsvg can load image with zero width or height. 

Well, but in that case the image has width/height zero, so any other attempt at loading it will also produce zero width/height. So... I'm not quite sure what should be done when a zero-sized image is loaded, except, well, having a zero sized image. Dunno....

> Is this approach ok?

Fine with me. Sorry if I am being annoying. :-)